arch/x86/core/intel64/irq.c

```diff
@@ -61,7 +60,9 @@ int z_x86_allocate_vector(unsigned int priority, int prev_vector)
 		vector = (priority * VECTORS_PER_PRIORITY) + IV_IRQS;
 	}
 
-	for (i = 0; i < VECTORS_PER_PRIORITY; ++i, ++vector) {
+	const int end_vector = vector + (int) VECTORS_PER_PRIORITY;
+
+	for (; vector < end_vector; ++vector) {
```

Comment:
No value is used after increment here; the comma operator doesn't inspect the value of its two subexpressions.